python/src/dataset.rs

```diff
@@ -449,6 +450,48 @@ impl Dataset {
         crate::arrow::record_batch_to_pyarrow(self_.py(), &batch)
     }
 
+    #[pyo3(signature = (row_slices, columns = None, batch_readahead = 10))]
+    fn take_scan(
```

Comment:
This is tested indirectly through `test_tf` but did you want to add one simple case testing this directly?

Reply:
I've added a test on the Rust side.